server/controllers/users/profile.js

```diff
@@ -50,6 +51,19 @@ export const update = async function(req, res) {
   if (sameEmail && sameEmail._id !== req.user._id)
     throw new BadRequestError('Email address is taken')
 
+  // Update admin status
+  if (includes(req.user, ADMIN_ROLE)) {
+    const alreadyAdmin = includes(user.roles, ADMIN_ROLE)
+    if (user.isAdmin && !alreadyAdmin) {
```

Comment:
I think this is pretty confusing. `req.user` is the original user, `user` is the original user updated with `req.body`, `isAdmin` is a prop set on the new user... I think it would be easier to follow if the admin role was added/removed on the client and the controller just compared hasRole(oldUser, admin) with hasRole(newUser, admin).

Sorry for the delay, I'll merge it like this because it does make sense, but I looked at it a couple of days ago and couldn't really follow it in the time i had so held off on it. It's really not bad though, just think it could be a bit clearer.